Document/0x05g-Testing-Network-Communication.md

```diff
@@ -169,6 +169,74 @@ OkHttpClient client = new OkHttpClient.Builder()
         .build();
 ```
 
+Applications that use WebView component may utilize event handler of the WebViewClient in order to perform some kind of "certificate pinning" on each request before that the target resource will be loaded. Verification code seems like the following:
+
+```java
+WebView myWebView = (WebView) findViewById(R.id.webview);
+myWebView.setWebViewClient(new WebViewClient(){
+    private String expectedIssuerDN = "CN=Let's Encrypt Authority X3,O=Let's Encrypt,C=US;";
+
+    @Override
+    public void onLoadResource(WebView view, String url)  {
+        //From Android API documentation about "WebView.getCertificate()":
+        //Gets the SSL certificate for the main top-level page or null if there is no certificate (the site is not secure).
+        //
+        //Available information on SslCertificate class are "Issuer DN", "Subject DN" and validity date helpers
+        SslCertificate serverCert = view.getCertificate();
+        if(serverCert != null){
+            //Apply check on Issuer DN against expected one
+            SslCertificate.DName issuerDN = serverCert.getIssuedBy();
+            if(!this.expectedIssuerDN.equals(issuerDN.toString())){
+                //Throw exception to cancel resource loading...
+            }
+        }
+    }
+});
+```
+
+Applications can decide to use the [Network Security Configuration](https://developer.android.com/training/articles/security-config.html) feature provided by Android from version 7.0.
+From Android documentation:
+"The Network Security Configuration feature lets apps customize their network security settings in a safe, declarative configuration file without modifying app code".
```

Comment:
I would remove this, as we don't want to just copy documentation and put a part of it in the sentence before instead. There you also have the reference.